stan/math/prim/fun/acosh.hpp

```diff
@@ -101,7 +101,7 @@ namespace internal {
  */
 template <typename V>
 inline std::complex<V> complex_acosh(const std::complex<V>& z) {
-  std::complex<double> y_d = acosh(value_of_rec(z));
+  std::complex<partials_type_t<V>> y_d = acosh(value_of(z));
```

Comment:
Why can't this operate on `double`?  We don't need the value, just the sign.  Is `value_of_rec` not pulling out the low-level `double` value?

Reply:
Oh sorry! I don't think I understood what was going on here so I used value_of instead. I'll switch this back.